api/lib/Cargo.toml

```diff
@@ -7,51 +7,51 @@ edition = "2021"
 workspace = true
 
 [dependencies]
-anyhow = "1.0"
-async-trait = "0.1.49"
-bs58 = "0.5.1"
-ed25519-dalek = "2.1.1"
-futures = "0.3.30"
-hex = "0.4.3"
-hmac-sha512 = "1.1.4"
-libsecp256k1 = { version = '0.7' }
-rand = "0.8.5"
-serde = { version = "1.0.197", features = ["derive"] }
-tiny-bip39 = "1.0.0"
-tokio = "1.28"
-tracing = "0.1"
-zeroize = "1.5.4"
-libp2p-identity = { version = "0.2", features = ["ed25519", "peerid"] }
-lazy_static = "1.4"
-jsonrpsee = { version = "0.23.2", features = ["full"] }
+anyhow = { workspace = true }
+async-trait = { workspace = true }
+bs58 = { workspace = true }
+ed25519-dalek = { workspace = true }
+futures = { workspace = true }
+hex = { workspace = true }
+hmac-sha512 = { workspace = true }
+libsecp256k1 = { workspace = true }
+rand = { workspace = true }
+serde = { workspace = true, features = ["derive"] }
+tiny-bip39 = { workspace = true }
+tokio = { workspace = true }
+tracing = { workspace = true }
+zeroize = { workspace = true }
+libp2p-identity = { workspace = true, features = ["ed25519", "peerid"] }
+lazy_static = { workspace = true }
+jsonrpsee = { workspace = true, features = ["full"] }
 
 # Local
-chainflip-engine = { path = "../../engine/" }
-utilities = { path = "../../utilities/" }
+chainflip-engine = { workspace = true }
+cf-utilities = { workspace = true }
 
 # State Chain
-cf-amm = { path = "../../state-chain/amm" }
-cf-chains = { path = "../../state-chain/chains" }
-cf-primitives = { path = "../../state-chain/primitives" }
-chainflip-node = { path = "../../state-chain/node" }
-custom-rpc = { path = "../../state-chain/custom-rpc" }
-pallet-cf-account-roles = { path = '../../state-chain/pallets/cf-account-roles' }
-pallet-cf-environment = { path = "../../state-chain/pallets/cf-environment" }
-pallet-cf-governance = { path = "../../state-chain/pallets/cf-governance" }
-pallet-cf-ingress-egress = { path = "../../state-chain/pallets/cf-ingress-egress" }
-pallet-cf-lp = { path = '../../state-chain/pallets/cf-lp' }
-pallet-cf-asset-balances = { path = '../../state-chain/pallets/cf-asset-balances' }
-pallet-cf-funding = { path = "../../state-chain/pallets/cf-funding" }
-pallet-cf-swapping = { path = '../../state-chain/pallets/cf-swapping' }
-pallet-cf-pools = { path = '../../state-chain/pallets/cf-pools' }
-pallet-cf-threshold-signature = { path = "../../state-chain/pallets/cf-threshold-signature" }
-pallet-cf-validator = { path = "../../state-chain/pallets/cf-validator" }
-state-chain-runtime = { path = "../../state-chain/runtime" }
+cf-amm = { workspace = true }
+cf-chains = { workspace = true }
+cf-primitives = { workspace = true }
+chainflip-node = { workspace = true }
+custom-rpc = { workspace = true }
+pallet-cf-account-roles = { workspace = true }
+pallet-cf-environment = { workspace = true }
```

Comment:
Strictly speaking, I think these should activate default features when it's in a std context (and to remain consistent with the previous dependency declaration). 

The way it's done in the polkadot repo is:

```
pallet-xx.workspace = true
pallet-xx.default-fetaures = true
```

I think we should follow the same convention.

This is true in a few places, not just here. For example in the node, the engine, etc. Anywhere were we previously *didn't* explicitly set `default-features = false` we now need to override our workspace default and set `default-features = true`.

Reply:
Right. 
I've added `default-features = true` for al the non- `no-std` crates, as well as dev-dependencies.